### PR TITLE
feat(rds): limitless database cluster

### PR DESCRIPTION
**Pull request by badmintoncryer**
### Issue # (if applicable)

None

### Reason for this change

Amazon Aurora has released Postgres Limitless Database and AWS CDK does not support it. 
<https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/limitless.html>

### Description of changes

- Add `clusterScailabilityType` prop to `clusterProps`
- Instance creation will only be executed during the creation of a standard Aurora cluster
- Add some validations
  - Requirements for limitless database is documented at [here.](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/limitless-reqs-limits.html)

### Description of how you validated changes

Add both unit and integ tests

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-rds/test/cluster.test.ts [hunk 1]
```diff
@@ -164,6 +166,260 @@ describe('cluster new api', () => {
       });
     });
 
+    test('cluster scailability option', () => {
```
Comment (mazyu36) [inline packages/aws-cdk-lib/aws-rds/test/cluster.test.ts:169]:
```suggestion
    test('cluster scalability option', () => {
```
nit: typo

**Review by mazyu36 — COMMENTED** [review]
Thank you for the contribution.
Just one minor comment.

**Comment by badmintoncryer** [comment]
@mazyu36 Thank you for your review! I've updated my code.

**Review by GavinZZ — APPROVED** [review]
LGTM, thank you for contributing!

**Comment by badmintoncryer** [comment]
@mergifyio update

**Comment by badmintoncryer** [comment]
@GavinZZ Thank you for your approval. It seems that codecov job has failed but I don't know how to resolve it. Could you please provide me with the solution?

**Comment by badmintoncryer** [comment]
@mergifyio update

**Pull request merged**